### DISCUSSION TITLE
Minimal standalone wasm binary

### DISCUSSION THREAD

**Discussion by yklcs** [sagemathinc/cowasm#64] [Q&A]
Hello, I'm interested in using `python-wasm` to power a Python repl in the browser. Currently, [`python-wasm` for the browser seems to be built into a rather complicated package](https://www.npmjs.com/package/python-wasm?activeTab=explore) by default. Is there any way to build a standalone wasm binary just for the repl? Something that just targets `wasm32-unknown-unknown` and exposes interpretation would be useful to those looking to build on `python-wasm`.

Thanks for all the excellent work.

**Comment by williamstein** [comment]
>  Is there any way to build a standalone wasm binary just for the repl?

Can you please clarify?  Do you just want something that is basically like the existing python-wasm package, but doesn't include numpy/pandas/sympy/etc. -- it contains only core Python packages that are included with Python?

**Reply by williamstein** [reply]
Also, to be clear, is this something you want to run via node.js on a server, or in a browser?

**Reply by yklcs** [reply]
I was hoping to build something for the browser with only the core Python interpreter and stdlib. I was wondering whether `dist/python.wasm` could be used (with some JS glue) to achieve this. The current build of `python-wasm` appears to load and bundle a lot of archives, which I want to avoid if at all possible.
